### PR TITLE
refactor(css/parser): many fixes

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Big refactor:
1. Refactor logic according spec https://www.w3.org/TR/css-syntax-3/#parser-algorithms
2. Don't try to parse value in declaration which previously has error, in this case we will output only one error (on the porblem places) and store the list of component values (that is why we reduced number of useless errors)
3. Better AST for custom properties
4. Fix bug with span when it contains spaces after `!important`
5. Fix parsing `!important`
6. A lot of small fixes
7. Better recovery mode

**BREAKING CHANGE:**

No

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_parser/tests/recovery/bad-url-token/double-quotes/output.json [hunk 1]
```diff
@@ -88,49 +88,23 @@
         },
         "value": [
           {
-            "type": "ListOfComponentValues",
+            "type": "Declaration",
```
Comment (alexander-akait) [inline crates/swc_css_parser/tests/recovery/bad-url-token/double-quotes/output.json:91]: Example of the better recovery mode - now we can parse it as declaration (due logic in spec)

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_css_parser

**Review by kdy1 — CHANGES_REQUESTED** [review]
clippy failed

**Comment by alexander-akait** [comment]
Yeah WIP on small regressions :+1:

**Review by kdy1 — APPROVED** [review]

**Pull request merged**